submariner-operator/move-subctl-to-its-own-repo.md

```diff
@@ -0,0 +1,69 @@
+# Move subctl to its own repository
+
+## Summary
+
+Currently, `subctl` code lives in the `submariner-operator` repository. Both `subctl` and the Operator require different sets of
+dependencies which might conflict with improving either or both. For example, `subctl` can't be `go install`-ed currently because the
+Operator dependencies don't allow it. It also makes Operator upgrades difficult.
+
+## Proposal
+
+The proposal is to move `subctl` out of the `submariner-operator` repository to its own repository. Everything specific to `subctl` will be
+moved to the new `subctl` repository. The `subctl` project will depend on the Operator (_e.g._ for the CRDs, RBAC etc. which are defined in
+the Operator). The `submariner-operator` repository will only contain artifacts which the Operator framework generates and/or needs.
+
+### Pros
+
+* Operator framework upgrades would be simpler.
+* It's logically correct to separate the Operator and the library as they are 2 different deliverables and provide different ways to
+  install Submariner.
+* Testing criterias are different. We don't need `subctl`-related GHAs for Operator and vice versa.
+* Development approach is different. One is a CLI and another is a framework.
+* Integration into `submariner-addon` might be simplified.
+
+### Cons
+
+* A new repository to maintain.
+* Splitting up the Operator and `subctl` will require care when implementing changes in either which affect the other.
+
+## Design details
+
+A new GitHub repository `subctl` will be created under the `submariner-io` organisation. The `submariner-operator` repository content will
+be copied to the new `subctl` repository, by pushing the git repository as-is, so that the full history is preserved. Non-`subctl`-related
+content will then be removed from the `subctl` repository, and non-Operator-related content will be removed from the `submariner-operator`
+repository. Once this is complete, there should be no duplicate content in the two repositories, other than the build framework. The
```

Comment:
```suggestion
repository. Once this is complete, there should be no duplicate code content in the two repositories. The
```

As-is, IMHO this would imply the license, code of conduct, and everything CI (linter config, GHA...) should not be duplicated.

Reply:
IMO, those thing go without saying. Each repo should have its own testing/GHAs. `other than the build framework` implies that only that part of the code could be duplicated.

Reply:
You have not addressed CI clearly enough in this EP though